src/compiler-debugging.md

```diff
@@ -18,6 +18,22 @@ Setting `debug = true` turns on many different debug options (e.g., `debug-asser
 `debug-logging`, etc.) which can be individually tweaked if you want to, but many people
 simply set `debug = true`. Check out the comments in config.toml.example for more info.
 
+If you want to use GDB to debug rustc, please set `config.toml` with options:
+
+```toml
+[rust]
+debug = true
+debuginfo-level = 2
```

Comment:
This should come with a big warning I think, I've found it to increase compile times and disk sizes a lot.

Reply:
Yes, it will increase disk size cost.
I will add a warning tip.
Do you know what the difference that's we don't set this option?
I may need to have a try.

Reply:
I haven't compared before and after, but I remember rustc_driver.so being over 3 GB (!!) with this enabled.

Reply:
emm, if we don't add `debuginfo-level = 2`, we can also follow the execution path with command `next`, `step`, but we can not print out the values.

With debuginfo we can print out those symbol info such as: 

```
(gdb) p callee_expr
$2 = (*mut rustc_hir::hir::Expr) 0x7fffe81512c8
(gdb) p callee_expr.kind
$3 = rustc_hir::hir::ExprKind::Path(rustc_hir::hir::QPath::Resolved(core::option::Option<&rustc_hir::hir::Ty>::None, 0x7fffe80513e8))
```

And, yes, those debug info will cost much more disk size and also the compiling speed.
For my case:

Disk size: 9G => 45G 
Time cost: 29 Minutes => 50 Minutes.

I will add more details in this PR. Anyway, even it costs so much, I think it's a helpful way to debug `rustc`, especially useful for beginners such as me.

Reply:
@jyn514
Added more details for the option.